bundle-workflow/tests/tests_test_workflow/test_test_args.py

```diff
@@ -0,0 +1,154 @@
+import logging
+import os
+import unittest
+from unittest.mock import patch
+
+from test_workflow.test_args import TestArgs
+
+
+class TestTestArgs(unittest.TestCase):
+
+    ARGS_PY = os.path.realpath(
+        os.path.join(os.path.dirname(__file__), "../../src/run_bwc_test.py")
+    )
+
+    @patch(
+        "argparse._sys.argv",
+        [
+            ARGS_PY,
+            "--s3-bucket",
+            "xyz",
+            "--opensearch-version",
+            "1.1.0",
+            "--build-id",
+            "30",
+            "--architecture",
+            "x64",
+            "--test-run-id",
+            "6",
+        ],
+    )
+    def test_required_arguments(self):
+        self.assertEqual(TestArgs().s3_bucket, "xyz")
+        self.assertEqual(TestArgs().opensearch_version, "1.1.0")
+        self.assertEqual(TestArgs().build_id, 30)
+        self.assertEqual(TestArgs().architecture, "x64")
+        self.assertEqual(TestArgs().test_run_id, 6)
+
+    @patch(
+        "argparse._sys.argv",
+        [
+            ARGS_PY,
+            "--s3-bucket",
+            "xyz",
+            "--opensearch-version",
+            "1.1.0",
+            "--build-id",
+            "30",
+            "--architecture",
+            "xyz",
+            "--test-run-id",
+            "6",
+        ],
+    )
+    def test_invalid_architecture(self):
+        with self.assertRaises(SystemExit):
+            self.assertEqual(TestArgs().architecture, "invalid")
+
+    @patch(
+        "argparse._sys.argv",
+        [
+            ARGS_PY,
+            "--s3-bucket",
+            "xyz",
+            "--opensearch-version",
+            "1111",
+            "--build-id",
+            "30",
+            "--architecture",
+            "x64",
+            "--test-run-id",
+            "6",
+        ],
+    )
+    def test_invalid_version(self):
+        with self.assertRaises(ValueError) as context:
+            self.assertEqual(TestArgs().opensearch_version, str(context.exception))
```

Comment:
This looks suspicious as the error raised in code is not just the version. You actually never compare the error message, it bails with an exception earlier. The comparison should be done outside of asserRaises. Note that in python scope of things inside with isn’t limited to it, so you can check context error after the with statement.

Reply:
Makes sense. Changed it